### DISCUSSION TITLE
global callback for error

### DISCUSSION THREAD

**Discussion by hossein-ghanbari** [TanStack/query#3025] [Q&A] [answered]
I use the following code to get errors globally.
in useQuery does work .
But in useMutation does not work

const queryClient = new QueryClient()

	queryClient.setDefaultOptions({
		queries: {
			onError: error => {
				console.log("Something went wrong:`");
			},
		},
		mutations: {
			onError: () => {
				console.log("Something went wrong:`");
			}
		}
	});

**Comment by TkDodo** [comment, marked as answer]
can you please provide a codesandbox reproduction showing that this doesn't work

**Reply by hossein-ghanbari** [reply]
My problem is solved, thank you

**Reply by TkDodo** [reply]
maybe leave an answer here for posterity please
